docs/block-validation/block-format.md

```diff
@@ -20,5 +20,54 @@ The FullBlock has fields for both the trunk and the foliage of the blockchain. T
 - **foliage_transaction_block**: Optional[FoliageTransactionBlock]: Transaction related metadata that is relevant for light clients (not actual transactions), only for tx blocks.
 - **transactions_info**: Optional[TransactionsInfo]: Transaction related metadata that is not relevant for light clients (not actual transactions), only for tx blocks.
-- **transactions_generator**: Optional[SerializedProgram]: A clvm program that generates all transactions (spends).
+- **transactions_generator**: Optional[SerializedProgram]: A clvm/rust program that generates all transactions (spends). See the next section for an important [update](#transactions_generator-update) due to the 2.1.0 hard fork.
 - **transactions_generator_ref_list**: List[uint32]: A list of block heights of previous generators referenced by this block's generator.
+
+## transactions_generator update
+
+Chia underwent a hard fork in version 2.1.0. This included updates to the `transactions_generator` code.
+
+### When
+
+The hard fork will activate at block `5 496 000`, which is expected to occur in June 2024. All wallets need to be compatible with the new implementation prior to this block.
+
+### Why
+
+Among other changes, the `transactions_generator` code was ported to Rust.
+
+There were multiple reasons for this update:
+- As an optimization -- Rust is generally more performant than Python
+- To support back refs
+- To make block validation faster
+- To enable compression with block refs by referencing subtrees of prior transactions
+
+### Where
+
+The code for these changes is held in two primary locations:
+- The [clvm_rs](https://github.com/Chia-Network/clvm_rs/blob/main/src/serde/de_br.rs) repo has the new serialization and deserialization code.
+- The [chia_rs](https://github.com/Chia-Network/chia_rs/tree/main/crates/chia-consensus/src/gen) repo has the consensus generator code.
+
+### What
+
+Two important changes went into this update:
+
+1. Allow serializing CLVM in a new, more compact form. This doesn't affect how CLVM is executed, it's just a matter of encoding. It does have some important consequences:
+    - Farmers can effectively stuff more transactions into blocks, because with a more compact encoding, you can fit more for the same byte-cost.
+    - The new implementation can take advantage of the de-duplication in the new serialization format, by caching tree-hashes. This effectively de-duplicates the work of hashing puzzles.
+
+    About the new serialization format:
+
+    The atom `0xfe` is followed by another atom, which is interpreted as a path into the environment (the same form as in CLVM). It references a node from a part of the tree that has already been deserialized (thus, allowing for de-duplicating sub-trees).
+
+2. The generator ROM implementation was ported from CLVM to Rust. This also doesn't affect the behavior of anything. It just speeds up block validation.
```

Comment:
for the purposes of this statement, the cost isn't considered behavior. The cost *does* change. This is explained below though, so it's probably OK.